crates/extrinsics/src/instantiate.rs

```diff
@@ -190,75 +307,51 @@ pub struct Exec {
     output_json: bool,
 }
 
-impl Exec {
-    async fn exec(&self, execute: bool) -> Result<(), ErrorVariant> {
+impl InstantiateExec {
+    /// Simulates the instantiation of a contract without executing it on the blockchain.
+    ///
+    /// This function performs a dry-run simulation of the contract instantiation process
+    /// and returns an [`InstantiateDryRunResult`] object containing essential
+    /// information, including the contract address, gas consumption, and storage
+    /// deposit.
+    ///
+    /// It does not modify the state of the blockchain.
+    pub async fn simulate_instantiation(
```

Comment:
I know there is already the `dry_run_instantiate` fn, but we should not introduce a new term here `simulate`, even if it is a correct synonym for `dry_run`. Find a way to unify these methods or rename this one. Should be similar to pattern in `call`

Reply:
This function essentially serves as a decoder for the result obtained from the `dry_run_instantiate` method. I've renamed it to `decode_instantiate_dry_run` and adjusted its parameter list to accept the result generated by `dry_run_instantiate`.